### DISCUSSION TITLE
Facebook picure / avatar url

### DISCUSSION THREAD

**Discussion by simpautus** [supabase/supabase#5408] [General]
I'm trying to get a proper user photo from Facebook login but since the profileURL in GoTrue is hard coded to `/me?fields=email,first_name,last_name,name,picture` it returns (default?) 50x50 url only. If I was able to alter the request URL to `me?fields=email,first_name,last_name,name,picture.width(1000)` it would return picture url to the nearest dimension to 1000px and so on. The default 50x50 is useless in my case. There's probably many other needs to tweak the request especially with wider permission scopes. Am I missing something or is there a workaround for this?

While we're at it I'd also need separate firstname and lastname from OAuth if available. They're provided by Facebook but joined together as `name`  by GoTrue.

**Comment by farzd** [comment]
i have the same issue, the id being returned is not the user id either, so cant make a graph request to fetch a photo either

**Comment by farzd** [comment]
the provider token you get from when you signin, you can use that to make another request 
![image](https://user-images.githubusercontent.com/1423413/177765834-fa38417c-2351-49d9-97ac-e2ffc0b7042e.png)

https://graph.facebook.com/me?fields=name,gender,location,picture.width(720).height(720)&access_token=xxxxxx

**Comment by farzd** [comment]
Actually this has sent me down a world of pain,
expo to standalone android app has issues when using oAuth, the returned response doesn't return a provider token or anything at all actually. [just this obj {type:dismiss}, even though it wasn't dimissed or cancelled ]
any chance we can change the size of the picture requested via goTrue?

https://github.com/expo/expo/issues/6679#

**Comment by Korayem** [comment]
Wouldn't running this query get you the facebook/google picture?

```sql
select identity_data::jsonb->>'picture' as picture from auth.identities
where provider='facebook' and user_id = '<user-id>'
```
